staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -614,11 +614,18 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 		if val == nil {
 			continue
 		}
-		structuralSchemas[v.Name], err = structuralschema.NewStructural(val.OpenAPIV3Schema)
+		s, err := structuralschema.NewStructural(val.OpenAPIV3Schema)
 		if *crd.Spec.PreserveUnknownFields == false && err != nil {
-			utilruntime.HandleError(err)
+			// This should never happen. If it does, it is a programming error.
+			utilruntime.HandleError(fmt.Errorf("failed to convert schema to structural: %v", err))
+			return nil, fmt.Errorf("the server could not properly serve the CR schema") // validation should avoid this
+		}
+		if err := structuraldefaulting.PruneDefaults(s); err != nil {
```

Comment:
should this be inside the `if *crd.Spec.PreserveUnknownFields == false` check?

Reply:
does it matter? We only allow defaults if `crd.Spec.PreserveUnknownFields == false`, and without default `PruneDefaults` is a noop.

Reply:
Double checked: either way, the integration tests stay green :+1: